ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala

```diff
@@ -39,16 +42,16 @@ private[kvutils] trait CommitContext {
 
   /** Set a value in the output state. */
   def set(key: DamlStateKey, value: DamlStateValue): Unit = {
-    if (!outputs.contains(key)) {
-      outputOrder += key
+    if (inputAlreadyContains(key, value)) {
+      logger.trace(s"Identical output found for key $key")
```

Comment:
I'd additionally remove the `key` from `outputs` and possibly `outputOrder`. Otherwise `set` has a weird semantics. For example,
```
inputs = Map(k1 -> Some(v1))

set(k1, v2)
set(k1, v1)
```
where `v1 != v2`. When we call `getOutputs`, we will get `Seq(k1 -> v2)` even though the second `set` should have overwritten the first. I would personally find it simpler if `set` remains unchanged and we only filter the outputs in `getOutputs`.

Reply:
Applied your suggestion (i.e., doing the filtration only in `.getOutputs`) and added a test case for verifying this corner case.